### PR TITLE
Automated cherry pick of #62481: kubeadm preflight: check socket path if defined otherwise

### PR DESCRIPTION
**Pull request by dixudx**
Cherry pick of #62481 on release-1.10.

#62481: kubeadm preflight: check socket path if defined otherwise

Fixes: kubernetes/kubeadm#814

### CODE REVIEW

**Comment by dixudx** [comment]
/assign  timothysc luxas 

@MikeSpreitzer @taharah FYI.

File: cmd/kubeadm/app/apis/kubeadm/types.go
Comment (luxas) [inline cmd/kubeadm/app/apis/kubeadm/types.go]: I haven't made up my mind yet whether this is a good or a bad abstraction at this level for the specific use case, but I can let this pass as a backport for the moment.

File: cmd/kubeadm/app/preflight/checks.go
Comment (luxas) [inline cmd/kubeadm/app/preflight/checks.go]: `kubeadmapiv1alpha1` in the future please

**Review by luxas — APPROVED** [review]
/lgtm
/approve

This is okay as a backport, but I think I want to rearrange some of the code at master.

**Comment by luxas** [comment]
/hold
for @detiber or @timothysc to sign off

**Comment by timothysc** [comment]
I really don't think this seems backport worthy.  We should only be backporting fixes that common bug scenarios, otherwise new changes will hit the next release cycle in ~1 month.

**Comment by MaciekPytel** [comment]
1.10 branch is now frozen for final testing before 1.10.3. Please don't remove hold on this until the release is cut (hopefully on Monday).

**Comment by timothysc** [comment]
I'm still of the opinion that this doesn't meet the criteria for a backport, fwiw.

**Comment by MaciekPytel** [comment]
Removing cherrypick-approved based on comment from @timothysc. I'm ok to reapply it once consensus is reached on whether this should be cherrypicked or closed.

/hold cancel
Since the branch is no longer frozen

**Comment by dixudx** [comment]
@luxas @timothysc Needs your LGTM again. Thanks.

**Comment by MaciekPytel** [comment]
@dixudx @timothysc @luxas Can you make a decision if this needs to be cherry-picked and lgtm or close this PR?

**Pull request closed without merging**
